Feature Request: `no-dupe-keys` covers classes.

## test.js

```js
/* eslint no-dupe-keys: 2 */
({ foo(){}, foo(){} });
class A { foo(){} foo(){} }
```

## Actual

```
> eslint -v
v1.0.0
> eslint test.js --env es6 --no-eslintrc

test.js
  2:2  error  Duplicate key 'foo'  no-dupe-keys

✖ 1 problem (1 error, 0 warnings)
```

## Expected

```
test.js
  2:2  error  Duplicate key 'foo'  no-dupe-keys
  3:1  error  Duplicate key 'foo'  no-dupe-keys

✖ 2 problem (2 error, 0 warnings)
```

Note: allows same names between static and non static.

--- a/conf/eslint.json
+++ b/conf/eslint.json
@@ -16,6 +16,7 @@
         "no-debugger": 2,
         "no-delete-var": 2,
         "no-div-regex": 0,
+        "no-dupe-class-members": 0,
         "no-dupe-keys": 2,
         "no-dupe-args": 2,
         "no-duplicate-case": 2,

--- a/docs/rules/README.md
+++ b/docs/rules/README.md
@@ -197,6 +197,7 @@ These rules are only relevant to ES6 environments.
 * [generator-star-spacing](generator-star-spacing.md) - enforce spacing around the `*` in generator functions
 * [no-class-assign](no-class-assign.md) - disallow modifying variables of class declarations
 * [no-const-assign](no-const-assign.md) - disallow modifying variables that are declared using `const`
+* [no-dupe-class-members](no-dupe-class-members.md) - disallow duplicate name in class members
 * [no-this-before-super](no-this-before-super.md) - disallow use of `this`/`super` before calling `super()` in constructors.
 * [no-var](no-var.md) - require `let` or `const` instead of `var`
 * [object-shorthand](object-shorthand.md) - require method and property shorthand syntax for object literals

--- a/docs/rules/no-dupe-class-members.md
+++ b/docs/rules/no-dupe-class-members.md
@@ -1,0 +1,62 @@
+# Disallow duplicate name in class members (no-dupe-class-members)
+
+If there are declarations of the same name in class members, the last declaration overwrites other declarations silently.
+It can cause unexpected behaviors.
+
+```js
+class Foo {
+  bar() { console.log("hello"); }
+  bar() { console.log("goodbye"); }
+}
+
+var foo = new Foo();
+foo.bar(); // goodbye
+```
+
+## Rule Details
+
+This rule is aimed to flag the use of duplicate names in class members.
+
+The following patterns are considered warnings:
+
+```js
+class Foo {
+  bar() { }
+  bar() { }
+}
+
+class Foo {
+  bar() { }
+  get bar() { }
+}
+
+class Foo {
+  static bar() { }
+  static bar() { }
+}
+```
+
+The following patterns are not considered warnings:
+
+```js
+class Foo {
+  bar() { }
+  qux() { }
+}
+
+class Foo {
+  get bar() { }
+  set bar(value) { }
+}
+
+class Foo {
+  static bar() { }
+  bar() { }
+}
+```
+
+## When Not to Use It
+
+This rule should not be used in ES3/5 environments.
+
+In ES2015 (ES6) or later, if you don't want to be notified about duplicate names in class members, you can safely disable this rule.

--- a/lib/rules/no-dupe-class-members.js
+++ b/lib/rules/no-dupe-class-members.js
@@ -1,0 +1,82 @@
+/**
+ * @fileoverview A rule to disallow duplicate name in class members.
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var stack = [];
+
+    /**
+     * Gets state of a given member name.
+     * @param {string} name - A name of a member.
+     * @param {boolean} isStatic - A flag which specifies that is a static member.
+     * @returns {object} A state of a given member name.
+     *   - retv.init {boolean} A flag which shows the name is declared as normal member.
+     *   - retv.get {boolean} A flag which shows the name is declared as getter.
+     *   - retv.set {boolean} A flag which shows the name is declared as setter.
+     */
+    function getState(name, isStatic) {
+        var stateMap = stack[stack.length - 1];
+        var key = "$" + name; // to avoid "__proto__".
+
+        if (stateMap[key] == null) {
+            stateMap[key] = {
+                nonStatic: {init: false, get: false, set: false},
+                static: {init: false, get: false, set: false}
+            };
+        }
+
+        return stateMap[key][isStatic ? "static" : "nonStatic"];
+    }
+
+    return {
+        // Initializes the stack of state of member declarations.
+        "Program": function() {
+            stack = [];
+        },
+
+        // Initializes state of member declarations for the class.
+        "ClassBody": function() {
+            stack.push(Object.create(null));
+        },
+
+        // Disposes the state for the class.
+        "ClassBody:exit": function() {
+            stack.pop();
+        },
+
+        // Reports the node if its name has been declared already.
+        "MethodDefinition": function(node) {
+            if (node.computed) {
+                return;
+            }
+
+            var name = node.key.name;
+            var state = getState(name, node.static);
+            var isDuplicate = false;
+            if (node.kind === "get") {
+                isDuplicate = (state.init || state.get);
+                state.get = true;
+            } else if (node.kind === "set") {
+                isDuplicate = (state.init || state.set);
+                state.set = true;
+            } else {
+                isDuplicate = (state.init || state.get || state.set);
+                state.init = true;
+            }
+
+            if (isDuplicate) {
+                context.report(node, "Duplicate name \"{{name}}\".", {name: name});
+            }
+        }
+    };
+};
+
+module.exports.schema = [];

--- a/tests/lib/rules/no-dupe-class-members.js
+++ b/tests/lib/rules/no-dupe-class-members.js
@@ -1,0 +1,75 @@
+/**
+ * @fileoverview Tests for no-dupe-class-members rule.
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/no-dupe-class-members");
+var RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+ruleTester.run("no-dupe-class-members", rule, {
+    valid: [
+        {code: "class A { foo() {} bar() {} }", ecmaFeatures: {classes: true}},
+        {code: "class A { static foo() {} foo() {} }", ecmaFeatures: {classes: true}},
+        {code: "class A { get foo() {} set foo(value) {} }", ecmaFeatures: {classes: true}},
+        {code: "class A { static foo() {} get foo() {} set foo(value) {} }", ecmaFeatures: {classes: true}},
+        {code: "class A { foo() { } } class B { foo() { } }", ecmaFeatures: {classes: true}},
+        {code: "class A { [foo]() {} foo() {} }", ecmaFeatures: {classes: true}}
+    ],
+    invalid: [
+        {
+            code: "class A { foo() {} foo() {} }",
+            ecmaFeatures: {classes: true},
+            errors: [
+                {type: "MethodDefinition", line: 1, column: 20, message: "Duplicate name \"foo\"."}
+            ]
+        },
+        {
+            code: "!class A { foo() {} foo() {} };",
+            ecmaFeatures: {classes: true},
+            errors: [
+                {type: "MethodDefinition", line: 1, column: 21, message: "Duplicate name \"foo\"."}
+            ]
+        },
+        {
+            code: "class A { foo() {} foo() {} foo() {} }",
+            ecmaFeatures: {classes: true},
+            errors: [
+                {type: "MethodDefinition", line: 1, column: 20, message: "Duplicate name \"foo\"."},
+                {type: "MethodDefinition", line: 1, column: 29, message: "Duplicate name \"foo\"."}
+            ]
+        },
+        {
+            code: "class A { static foo() {} static foo() {} }",
+            ecmaFeatures: {classes: true},
+            errors: [
+                {type: "MethodDefinition", line: 1, column: 27, message: "Duplicate name \"foo\"."}
+            ]
+        },
+        {
+            code: "class A { foo() {} get foo() {} }",
+            ecmaFeatures: {classes: true},
+            errors: [
+                {type: "MethodDefinition", line: 1, column: 20, message: "Duplicate name \"foo\"."}
+            ]
+        },
+        {
+            code: "class A { set foo(value) {} foo() {} }",
+            ecmaFeatures: {classes: true},
+            errors: [
+                {type: "MethodDefinition", line: 1, column: 29, message: "Duplicate name \"foo\"."}
+            ]
+        }
+    ]
+});

